### PR TITLE
Implemented SkipWhile

### PR DESCRIPTION
**Pull request by nikolalukovic**
#22 Implementation

### CODE REVIEW

**Comment by nikolalukovic** [comment]
Concerning the CRLF endings requierment, i'd recommend to add .gitattributes and set those rules for this project. My git settings convert crlf to lf by default, i'd really not like to change my git global settings because of this.

File: src/Enumerables.ts [hunk 1]
```diff
@@ -801,6 +806,46 @@ export class ConditionalEnumerable<TElement> extends Enumerable<TElement>
...
+        super(source);
+        this._predicate = predicate;
+        this._shouldContinueChecking = true;
+    }
+
+    public copy(): SkipWhileEnumerable<TElement>
+    {
+        return new SkipWhileEnumerable<TElement>(this.source.copy(), this._predicate);
+    }
+
```
Comment (isc30) [inline src/Enumerables.ts]: Override reset to also set `_shouldContinueChecking` to `true`

File: test/unitary/IQueryable.test.ts [hunk 2]
```diff
@@ -1166,6 +1167,49 @@ export namespace IQueryableUnitTest
         });
     }
 
+    function skipWhile(instancer: Instancer): void
+    {
+        it("Empty if empty", () =>
+        {
+            const base = instancer([]);
+            Test.isArrayEqual(base.skipWhile(e => true).toArray(), []);
```
Comment (isc30) [inline test/unitary/IQueryable.test.ts]: also check that `false` predicate in empty enumerable returns empty

File: src/Collections.ts [hunk 3]
```diff
@@ -209,6 +209,11 @@ export abstract class EnumerableCollection<TElement>
         return this.asEnumerable().singleOrDefault();
     }
 
+    public skipWhile(predicate: Predicate<TElement>): IEnumerable<TElement>
```
Comment (isc30) [inline src/Collections.ts]: add loop based implementation in `ArrayEnumerable` (I just noticed that implementation for `where` is also missing (`filter` will do the trick here))

Comment (nikolalukovic) [reply src/Collections.ts]: Hmm, i'm not sure what i need to do here. Do i add `skipWhile` method to `ArrayEnumerable`? If that's the case why can't i do it like this?

    public skipWhile(predicate: Predicate<TOut>): IEnumerable<TOut>
    {
        return this.list.skipWhile(predicate);
    }

File: src/Enumerables.ts [hunk 4]
```diff
@@ -820,6 +820,11 @@ export class SkipWhileEnumerable<TElement> extends Enumerable<TElement>
         this._shouldContinueChecking = true;
     }
 
+    public reset(): void
+    {
+        this._shouldContinueChecking = true;
```
Comment (isc30) [inline src/Enumerables.ts]: also call `super.reset();` before

File: src/Collections.ts (hunk 3 above)
Comment (isc30) [reply src/Collections.ts]: oh sorry, my fault, his doesn't affect deferred-executed methods. The idea behind `ArrayEnumerable` is to create optimized versions of the non-deferred algorithms. Not this case.

**Comment by isc30** [comment]
Looks like [this line](https://coveralls.io/builds/15778784/source?filename=src%2FEnumerables.ts#L830) is missing coverage. This can be achieved by adding `SkipWhileEnumerable` to `IQueryable.test.ts@runTest`

**Comment by isc30** [comment]
perfect! thanks a lot for the contribution!

**Pull request merged**